core/modules/replica/AddReplicaQservMgtRequest.cc

```diff
@@ -138,24 +136,42 @@ void AddReplicaQservMgtRequest::startImpl(util::Lock const& lock) {
 
 void AddReplicaQservMgtRequest::finishImpl(util::Lock const& lock) {
 
-    if (extendedState() == ExtendedState::CANCELLED) {
+    switch (extendedState()) {
 
-        // And if the SSI request is still around then tell it to stop
+        case ExtendedState::CANCELLED:
+        case ExtendedState::TIMEOUT_EXPIRED:
 
-        if (_qservRequest) {
-            bool const cancel = true;
-            _qservRequest->Finished(cancel);
-        }
+            // And if the SSI request is still around then tell it to stop
+
+            if (_qservRequest) {
+                bool const cancel = true;
+                _qservRequest->Finished(cancel);
+            }
+            break;
+
+        default:
+            break;
```

Comment:
other states are happily ignored? or are unexpected? (if the latter I think you should log or somehow warn here)

Reply:
This is XRootD/SSI protocol which requires to explicitly notify a remote server on an abnormal condition when a client-side request is being aborted after it's started and before it's finished. The client requests get explicitly cancelled or get expired due to a timeout. Hence these two are mentioned in the code. For others the server cancellation is not needed.
I may still need to talk to @abh3 to see if I'm not missing anything here.